src/pages/CityPage/CityPageGuides.jsx

```diff
@@ -1,61 +1,151 @@
-import { useState, memo } from 'react';
+import { useState, useEffect, memo } from 'react';
 import { Link } from 'react-router-dom';
+import { ErrorBoundary } from 'react-error-boundary';
 
-import GuideCard from '../../components/common/GuideCard/GuideCard';
 import CategoryCard from '../../components/common/CategoriesCards/CategoryCard';
 import CreateGuideButton from '../../components/common/CreateGuideButton/CreateGuideButton';
+import GuidesList from '../../components/common/GuidesList/GuidesList';
+import Icon from '../../components/common/Icon/Icon';
+import CategoriesSkeleton from '../../components/common/CategoriesCards/CategoriesSkeleton';
 
-import CategoriesButtonsData, { Icons } from '../../data/categoriesData';
-import Stub from '../../assets/img/cities/Stub.png';
+import useAllGuides from '../../hooks/useAllGuides';
+import { fetchCategories } from '../../services/CategoriesService';
+
+import FallbackIcon from '../../utilities/FallbackIcon';
+import NoCategory from '../../assets/img/CategoriesCards/NoCategory.png';
 
 const CityPageGuides = () => {
   const [activeKeys, setActiveKeys] = useState([]);
+  const [categories, setCategories] = useState([]);
+  const [loadingCategories, setLoadingCategories] = useState(true);
+  const [errorCategories, setErrorCategories] = useState(null);
+
+  useEffect(() => {
+    const getCategories = async () => {
+      try {
+        const categoriesData = await fetchCategories();
```

Comment:
this should be called under `useQuery` to get such benefits as caching, refetching etc

Reply:
Great idea! I'll improve it and create a separate hook!
Initially, I thought that we would need guide categories only in 1 place and we could create the functionality for receiving them directly in the component. But due to the amount of code, I forgot the principle of separating logic, that the CityPageGuides component is engaged in displaying the interface, and not the logic of receiving data.